src/resources.js

```diff
@@ -428,6 +463,7 @@ export class Resources {
         ', prerenderSize=', this.prerenderSize_);
     this.pass_.cancel();
     this.vsyncScheduled_ = false;
+    this.focusHistory_.purgeBefore(now - FOCUS_HISTORY_TIMEOUT_);
```

Comment:
The logic currently is allowed for all AMP elements. amp-twitter and amp-iframe are first customers. Generally I wanted to support amp-iframe, not just iframe since there could, in the future, be some embellishments in these elements that wall under this logic as well. To make it simpler, though, I can definitely move purge inside the FocusHistory itself.